corehq/apps/reports/static/reports/js/filters/case_list_explorer.js

```diff
@@ -37,16 +37,27 @@ hqDefine("reports/js/filters/case_list_explorer", ['jquery', 'underscore', 'knoc
         self.name = ko.observable(name).trimmed();
         self.label = ko.observable(label).trimmed();
 
-        self.meta_type = ko.computed(function () {
-            var value = _.find($parent.allSuggestions, function (prop) {
+        self._value = function () {
+            return _.find($parent.allSuggestions, function (prop) {
                 return prop.name === self.name();
             });
+        };
+
+        self.meta_type = ko.computed(() => {
+            var value = self._value();
             if (value) {
                 return value.meta_type;
             }
             return null;
         });
 
+        self.name.subscribe(() => {
+            var value = self._value();
+            if (value) {
+                self.label(value.label || "");
+            }
+        });
```

Comment:
what does this section do?

Reply:
this section updates the label when a user selects a case_property.

Reply:
Okay. I was confused by the method extraction, `_value`.